src/com/scottyplunkett/server/HTTPRequest.java

```diff
@@ -3,25 +3,52 @@
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Collections;
 
 class HTTPRequest {
     private ArrayList<String> requestContent;
     private String requestLine;
+    private String eTag;
+    private String body;
 
     HTTPRequest(BufferedReader in) throws IOException {
         requestContent = new ArrayList<>();
-        String s;
-        while ((s = in.readLine()) != null) {
-            requestContent.add(s);
-            if (s.isEmpty()) {
+        StringBuilder fullRequest = new StringBuilder();
+        int nextChar;
+        while ((nextChar = in.read()) != -1) {
+            fullRequest.append((char)nextChar);
+            if (!in.ready()) {
                 break;
             }
         }
+        Collections.addAll(requestContent, fullRequest.toString().split("\r\n"));
         requestLine = requestContent.get(0);
+        setEtag();
+        setBody();
+    }
+
+    private void setEtag() {
+        String secondLineHeader = requestContent.get(1).split("\\s")[0];
```

Comment:
These splits may be a good refactoring opportunity, give what is at that index a bit more meaning within an object.